content/blog/2018/06/2018-05-14-simple-pull-request-plugin.adoc

```diff
@@ -0,0 +1,166 @@
+---
+layout: post
+title: "GSoC Project Intro: Simple pull request Plugin"
+tags:
+- gsoc2018
+- plugin
+author: abhishek_gautam
+---
+
+=== About me
+
+I am Abhishek Gautam, 3rd year student from Visvesvaraya National Institute of
+technology, India, Nagpur. I was a member of ACM Chapter and Google student developer club of his
+college. I am passianate about automation.
+
+=== Mentors
+
+* link:https://github.com/oleg-nenashev[Oleg Nenashev] (Org Admin)
+* link:https://github.com/martinda[Martin d'Anjou]
+* link:https://github.com/kwhetstone[Kristin Whetstone]
+* link:https://github.com/grandvizier[Jeff Knurek]
+
+=== Project Summary
+
+This is a GSoC 2018 project.
+
+This project aims to develop a Pull request Job Plugin. Users should be able to
+configure job type using YAML file which will be placed in root directory of the
+Git repository being the subject of the PR. Plugin should interact with various
+platfoms like Bitbucket, Github, Gitlab, etc whenever a pull request is created or updated.
+
+Plugin should be able to detect the presence of certain types of the report at conventional locations,
+and publish them automatically. If the reports are not in a conventional location,
+users should be able to specify the location using the YAML file.
+
+==== Benefits to the community
+
+* Project administrators will be able to handle builds for pull requests more easily.
+* Build specifications for pull request can be written in a more readable format.
+* Build reports will be automatically published to Github, Bitbucket, ect.
+* Build status updates will be sent to git servers automatically.
+* Users will not have to deal with pipeline code.
+
+=== Prior work
+
+. link:https://github.com/jenkinsci/travis-yml-plugin[Travis YML Plugin]: 
+  Designed to run .travis.yml as jenkins pipeline job.
+  Travis-CI does not support external pull requests hence. And jenkins environment
+  is different than Travis and does not always make sense to use configurations
+  defined for other environment in jenkins. Also maintenance of this is slowed
+  down and last commit for this plugin was on 14 Nov 2016.
+  https://github.com/jenkinsci/travis-yml-plugin/commits/master[Click here] to check.
+
+. link:https://github.com/ndeloof/codeship-plugin[CodeShip Plugin]: 
+  This plugin is designed to convert codeship "steps.yaml" and
+  "services.yaml" to scripted pipeline code. This plugin has never been released.
+
+
+
+=== Design
+
+This plugin will be developed on the top of the MultiBranch Pipeline plugin.
+
+For now the plugin is bulding branches and Pull request both using Jenkinsfile.yaml,
```

Comment:
Would the specific comments about building both branches and pull requests and fixing in the next coding phase fit better under the "Implemented as of 2018 June 13" section?  Based on the suggestions to keep the above sections in present tense, I think that makes sense.

Reply:
I agree. Using the present tense helps with readability. Even the "wish list" can be written in the present tense, as long as the section title is clear that the section content is about declaring the intent or the goals.